html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/data-url-agent-cluster.https.html

```diff
@@ -0,0 +1,45 @@
+<!doctype html>
+<meta charset=utf-8>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/wasm/jsapi/wasm-module-builder.js"></script>
+<div id=log></div>
+<script>
+const dataURLWorker = `
+  onmessage = () => self.postMessage("FAIL");
+  onmessageerror = () => self.postMessage("PASS");
+`;
+
+const dataFrameScript = `
+  onmessage = () => parent.postMessage("FAIL", "*");
+  onmessageerror = () => parent.postMessage("FAIL", "*");
+`;
+
+[
+  new SharedArrayBuffer(),
+  new WebAssembly.Module(new WasmModuleBuilder().toBuffer())
+].forEach(instance => {
+  async_test(t => {
+    const worker = new Worker(`data:,${dataURLWorker}`);
+    worker.onmessage = t.step_func_done(({ data }) => {
+      assert_equals(data, "PASS");
+    });
+    worker.postMessage(instance);
+  }, `As data URL dedicated workers form their own agent cluster, ${instance.constructor.name} cannot be messaged to them`);
+
+  promise_test(t => {
```

Comment:
This seems like it'd be simpler as an `async_test`, like the above. But, I guess you need to use `promise_test` since everyone's sharing a `message` event listener. So maybe add a comment to that effect.